1.16.2 Release

aikar opened the pull request:
ENSURE YOU HAVE A BACKUP OF ALL WORLDS BEFORE STARTING YOUR SERVER WITH 1.16.2 BUILDS

Creating as PR so can merge as squash but retain history for changes

latest 1.16.2 build prob ready for release - note there is a NEW issue of the "MapLike" data conversions spam. 

https://bugs.mojang.com/browse/MC-197883

UNSURE OF DATA INTEGRITY - Please test and ensure your Villages, Pillager Outposts and Bastions are converting from old versions correctly. This includes 1.16.1 too.


ENSURE YOU HAVE A BACKUP OF ALL WORLDS BEFORE STARTING YOUR SERVER WITH 1.16.2 BUILDS


ENSURE YOU HAVE A BACKUP OF ALL WORLDS BEFORE STARTING YOUR SERVER WITH 1.16.2 BUILDS


ENSURE YOU HAVE A BACKUP OF ALL WORLDS BEFORE STARTING YOUR SERVER WITH 1.16.2 BUILDS

kennytv committed:
Add smithing item helpers

aikar committed:
Current progress on 1.16.2

BillyGalbreath committed:
Almost ready

Proximyst committed:
It compiles!

BillyGalbreath committed:
Oopsies

BillyGalbreath committed:
volatile

BillyGalbreath committed:
missed this

BillyGalbreath committed:
Add an OBFHELPER

aikar committed:
Merge pull request #4112 from BillyGalbreath/progress/1.16.2

Proximyst committed:
Fix running; forgot to push this

Proximyst committed:
Fix test, add strider mob goal and fix piglin goals

Proximyst committed:
Updated Upstream (Bukkit/CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
b2e0ded3 SPIGOT-6065: Allow empty AttributeModifier name

CraftBukkit Changes:
43c7ff9d Fix nether portal ignition
2ee7318a SPIGOT-6064: world.getBiome() null Error
9f12e29b Fix crash with composters
22c3a2f9 Fix crash growing saplings
62761be2 SPIGOT-6061: NPE in CraftBlockTag.getHandle()

Proximyst committed:
Import fastutil classes patch

krolik-exe committed:
Fix IDE debug JVM flag for 1.16.2

krolik-exe committed:
Add this to SHIT_TO_CHECK.md aswell

aikar committed:
Fix anti xray

aikar committed:
Updated Upstream (Bukkit/CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
1fc6268e SPIGOT-4705: Revamp Ageable interface and add Breedable interface
e9c37513 #536: Add missing ItemFlag.HIDE_DYE

CraftBukkit Changes:
f26d8e84 SPIGOT-4705: Revamp Ageable interface and add Breedable interface

commandz0 committed:
Improve handling of indestructable blocks

Proximyst committed:
Don't mark null chunk sections for block updates

Proximyst committed:
Remove armour stand double add to world

Proximyst committed:
Initialise a new chunk section if none was found yet updated

BillyGalbreath committed:
Fix MC-187716 Use configured height

BillyGalbreath committed:
Fix incorrect return for WorldServer#addAllEntitiesSafely (#3)

aikar committed:
Merge pull request #4132 from Proximyst/progress/1.16.2

aikar committed:
Updated Upstream (Bukkit/CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
bf0015eb #540: Add Inventory#isEmpty()
c45e2faf #539: HumanEntity - add a method to make the entity drop the item they're holding
61303b47 #537: Add EntitySpellCastEvent for spellcaster illager
05b2dc61 #538: Un-deprecate now implemented player portal event api

CraftBukkit Changes:
e850144b #726: Add Inventory#isEmpty()
13945b76 #725: HumanEntity - add a method to make the entity drop the item they're holding
82542d01 #723: Implement EntitySpellCastEvent for spellcaster illager
1e76d09f #724: Re-implement player portal event search radius and creation API

aikar committed:
Merge remote-tracking branch 'origin/master' into progress/1.16.2

* origin/master:
  Fix regex mistake in CB NBT int deserialization (#4146)

aikar committed:
Updated Upstream (CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

CraftBukkit Changes:
419018f5 SPIGOT-6075: Snowballs don't destroy Ender Crystals
fa917c47 Don't create custom registry for custom worlds
e354566e SPIGOT-6076: Crash when updating blocks in uninitialized chunk sections
5a66916c Cleanup sendBlockChange

stonar96 committed:
Do not obfuscate air (#4149)

electronicboy committed:
Updated Upstream (Bukkit/CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
18e9d931 #543: Let World#setSpawnLocation() support an angle

CraftBukkit Changes:
eaf0bace Increase outdated build delay
8abed406 SPIGOT-6086: Twisting vines growing at top of world crashes server
59a0bafa SPIGOT-6091: End platform does not generate
9a9c2234 SPIGOT-6087: Revert "SPIGOT-6086: Twisting vines growing at top of world crashes server"
2e178b09 #730: Let World#setSpawnLocation() support an angle
fad5fb9e SPIGOT-6086: Twisting vines growing at top of world crashes server

jpenilla committed:
Fix MC-99259 Wither Boss Bar doesn't update until invulnerability period is over (#4164)

This fix was as simple as suggested by the code review on the issue at Mojira https://bugs.mojang.com/browse/MC-99259

I simply moved the call outside of the if/else block so it is called whether the Wither is invulnerable or not

aikar committed:
Merge remote-tracking branch 'origin/master' into progress/1.16.2

* origin/master:
  [CI-SKIP] Create CODEOWNERS

krolik-exe committed:
Fix IDE Debug JVM Flag 2 (#4141)

aikar committed:
update checkout-pr to latest I use

aurorasmiles committed:
[CI-SKIP] Update issue templates with notices

Add comment about our support to older versions and forks of Paper.

Proximyst committed:
[CI-SKIP] Make checkoutpr.sh use $SHELL (#4167)

Commit:
Fix MC-197883 by specifying the typeKey

electronicboy committed:
Decomp fix

Proximyst committed:
Don't require FACING data

Fixes PaperMC/Paper#3084.

aikar committed:
Updated Upstream (CraftBukkit)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

CraftBukkit Changes:
fe4efd19 SPIGOT-6093: Can't change ItemStack in EntityPickupItemEvent

aikar committed:
Merge branch 'master' into progress/1.16.2

* master:
  Do not let the server load chunks from newer versions (1.16.2+)

aikar committed:
MC-197883: Bandaid decode issue

Mojang has a mix of type and name in the data sets, but you can only
use one.

This will retry as name if type is asked for and not found.

Proximyst committed:
Fix SpawnChangeEvent not firing for all use-cases

Fixes PaperMC/Paper#1913.

Proximyst committed:
Add #setMaxPlayers API

Closes PaperMC/Paper#2028.

Proximyst committed:
Fix MC-197271

Commit:
Brand support

BillyGalbreath committed:
Add moon phase API

Proximyst committed:
[CI-SKIP] Require dependencies script (#4172)

BillyGalbreath commented on Spigot-API-Patches/0218-Add-setMaxPlayers-API.patch:
```diff
@@ -0,0 +1,48 @@
++    /**
++     * Set the maximum amount of players which can login to this server.
++     *
++     * @param the amount of players this server allows
```
`maxPlayers` missing in `@param` (makes Jenkins fail builds when creating Javadocs)

Proximyst replied:
I *almost* did it right! Easy fix though, feel free to PR, or I'll do it once I'm available tomorrow.

BillyGalbreath commented on Spigot-API-Patches/0218-Add-setMaxPlayers-API.patch:
```diff
@@ -0,0 +1,48 @@
++    /**
++     * Set the maximum amount of players which can login to this server.
++     *
++     * @param the amount of players this server allows
```
`maxPlayers` missing in `@param` (makes Jenkins fail builds when creating Javadocs)

BillyGalbreath committed:
Fix javadocs failing jenkins builds (#4189)

electronicboy committed:
Fix NPE when calling Player#getClientBrandName too early

electronicboy committed:
Fix x derp

electronicboy committed:
Rebuild patches

ishland committed:
Repull MC-197271 patch manually

Derp on my behalf appears to have made git upset, just going to
reshove this patch in manually and pray

electronicboy committed:
Updated Upstream (Bukkit/CraftBukkit/Spigot)

Upstream has released updates that appears to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
29e9b367 SPIGOT-974: Add ArmorStand locking API

CraftBukkit Changes:
bad55dbb SPIGOT-6097: Zoglins not attacking mobs
850e7b14 SPIGOT-974: Add ArmorStand locking API
7d8884b2 Also make field volatile
50da2cde SPIGOT-6094: Error loading datapack functions
8b52a766 SPIGOT-6092: Player#setBedSpawnLocation - NullPointerException

Spigot Changes:
d65430a1 Drop json plugins command as it cannot handle large amounts of plugins

Proximyst committed:
[CI-SKIP] Require /version, not build num

aikar committed:
Rebuild patches

aikar committed:
Let's try updating git since travis is out of date

electronicboy committed:
Update travis.yml

electronicboy committed:
Bump travis back to trusty because java 8

aikar committed:
update apt before upgrade

electronicboy committed:
Bump back to focal, disable java 8 for "now"

aikar committed:
remove unnecessary apt update for git
